clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -865,6 +886,8 @@ public void onAssignment(Set<TopicPartition> assignment) {
         private final TopicPartition partition;
         private final CompletedFetch completedFetch;
         private final Iterator<? extends RecordBatch> batches;
+        private final Set<Long> abortedPids;
+        private final Queue<FetchResponse.AbortedTransaction> abortedTransactions;
```

Comment:
It helps ensuring that the right requests go in the right order when there are multiple transactional  requests in flight. For instance, we could have no valid coordinators, pending `AddPartitionsToTransaction` requests, and an offset commit request. In this case, we would have two `FindCoordinatorRequests` (one at a time), one or more `AddPartitionToTransactionRequest`, and a `AddOffsetsToTransactionRequest` in the queue. With the priority queue, we are guaranteed that the find coordinator requests will go first. 

Conversely, it also ensures that the `EndTransactionRequest` goes last.